backend/test_firestore.go

```diff
@@ -52,3 +52,34 @@ func GetData(w http.ResponseWriter, r *http.Request) {
 	// marshal the data struct to JSON to send as response
 	json.NewEncoder(w).Encode(user)
 }
+
+func DeleteData(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	if vars["id"] == "" {
+		log.Fatal("no document id provided")
+	}
+	user := &firebase.User{FirebaseID: vars["id"]}
+	err := user.Delete()
+	if err != nil {
+		log.Fatalf("failed to delete user : %v", user)
+	}
+	w.WriteHeader(http.StatusNoContent)
+}
+
+
+func UpdateUser(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	if vars["id"] == "" {
+		log.Fatal("no document id provided")
+	}
+	var user firebase.User
+	user.FirebaseID = vars["id"]
```

Comment:
is there a reason you chose to do
```go 
var user firebase.User
user.FirebaseID = vars["id"]
```

instead of 

```go
user := &firebase.User{FirebaseID: vars["id"]}
```